application/src/Controller/Admin/VocabularyController.php

```diff
@@ -115,26 +125,112 @@ public function editAction()
         $data = $vocabulary->jsonSerialize();
         $form->setData($data);
 
+        $view = new ViewModel;
+        $view->setVariable('vocabulary', $vocabulary);
+
         if ($this->getRequest()->isPost()) {
             $data = $this->params()->fromPost();
             $form->setData($data);
             if ($form->isValid()) {
                 $response = $this->api($form)->update('vocabularies', $this->params('id'), $data, [], true);
                 if ($response->isSuccess()) {
-                    $this->messenger()->addSuccess('Vocabulary successfully updated'); // @translate
-                    return $this->redirect()->toRoute(null, ['action' => 'browse'], true);
+                    $fileData = $this->params()->fromFiles('file');
+                    if (0 === $fileData['error']) {
+                        $this->messenger()->addSuccess('Vocabulary label and comment successfully updated. Please review these changes before you accept them.'); // @translate
```

Comment:
I feel like this message is possibly confusing for the usual case where people don't actually update the label and comment. The fact that those changes happen immediately but these go on a delay is just tough to convey clearly.